website/docs/concepts/provider_lifecycles.mdx

```diff
@@ -0,0 +1,102 @@
+---
+title: Provider Lifecycles
+---
+
+import CodeBlock from "@theme/CodeBlock";
+import onDispose from "!!raw-loader!/docs/concepts/lifecycle_on_dispose.dart";
+import { trimSnippet } from "../../src/components/CodeSnippet";
+
+## When does my Provider get created and disposed?
+
+The states that all different types of providers can go through are the same:
+
+- Uninitialized
+- Alive
+- Paused
+- Disposed
+
+### Disposed / Uninitialized
+
+An **Uninitialized** or **Disposed** provider does not take up any memory since its state is not initialized.
+Essentially it is just a definition of how to create the provider's state when you need it.
+It will stay that way until an **Alive** provider or a [WidgetRef] from the UI reads, watches, or listens to it.
+
+### Creating -> Alive
+
+When an **Uninitialized** provider is read, listened to or watched it's state will be created.
+
+During creation your provider's build function will be run. 
+Any providers that you read or watch using the `ref` exposed by the callback will be created as needed and their state will be retrieved.
+
+If there are any circular dependencies during this creation process Riverpod will throw an error.
+The best way to fix this error is to redesign your dependencies to have a uni-directional dataflow.
+
+The provider's state is stored in a [ProviderContainer]. In a Flutter app this container is in a [ProviderScope] widget.
+
+As such, even though the definition of how to create the state (the provider) is global, the state is actually local,
+and can be different in different portions of your UI using nested [ProviderScope] widgets and overrides.
+
+This is very similar to how flutter widgets work. You only pay for the definition once, but can reuse the state in different parts of the tree as needed.
+
+### Alive
+
+When your provider is **Alive**, changes to its state will cause dependent providers and/or the dependent UI to rebuild.
+
+From the other perspective, as a reactive framework, you can watch other providers to have the provider recreate itself whenever one of it's dependencies changes.
+
+If you need to have some long-lived state that depends on other state you can use Ref's [listen] method to subscribe for changes on another provider without causing a rebuild of the provider.
+
+If you need to use the state from another provider in a side-effect, you can use Ref's [read] method to obtain the current state from another provider.
+
+Typically when constructing a [StateNotifier] or [ChangeNotifier] class you should pass in the `ref` to allow the Notifier to obtain the current value of dependencies as needed. By using the new [Notifier] and [AsyncNotifier] classes from Riverpod 2.0, the ref is already available as an instance member of the class.
+
+### Alive -> Paused
+When an **Alive** provider is no longer listened to by other providers or the UI, it goes into a **Paused** state. 
+This means that it no longer will react to changes on providers it is listening to.
```

Comment:
This seems like very important information, and also raises a lot of questions for me about the behavior of a paused, then unpaused provider.

What happens when the `provider` becomes unpaused? Will it be updated with the most recent state of the providers it is listening to? Won't e.g. a `StateNotifier` that uses the `ref` of a `StateNotifierProvider` potentially miss out on state-changing events that it is listening to?

Reply:
I could be wrong, but I believe that it will still be marked 'dirty', and will be reinitialized if anything it was watching had changed while it was paused. I'm not entirely sure about listeners @rrousselGit, am I explaining paused providers right?